client/block.c

```diff
@@ -559,6 +640,50 @@ static int add_block_nolock(struct xdag_block *newBlock, xdag_time_t limit)
 		}
 	}
 
+#if CACHE == 1
```

Comment:
I suggest to move that part and next one to a separate functions and leave here
```
#if CACHE
recalculate_cache(...);
#endif
```
You can choose another name for function.